.github/workflows/clippy_dev.yml

```diff
@@ -21,35 +21,49 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    # Setup
-    - name: Checkout
-      uses: actions/checkout@v2.3.3
-
-    - name: remove toolchain file
-      run: rm rust-toolchain
-
-    - name: rust-toolchain
-      uses: actions-rs/toolchain@v1.0.6
-      with:
-        toolchain: nightly
-        target: x86_64-unknown-linux-gnu
-        profile: minimal
-        components: rustfmt
-        default: true
-
-    # Run
-    - name: Build
-      run: cargo build --features deny-warnings
-      working-directory: clippy_dev
-
-    - name: Test limit_stderr_length
-      run: cargo dev limit_stderr_length
-
-    - name: Test update_lints
-      run: cargo dev update_lints --check
-
-    - name: Test fmt
-      run: cargo dev fmt --check
+      # Setup
+      - name: Checkout
+        uses: actions/checkout@v2.3.3
+
+      # To test formatting, we need to temporarily disable the rust-toolchain file
+      - name: remove toolchain file
+        run: rm rust-toolchain
+
+      - name: rust-toolchain
+        uses: actions-rs/toolchain@v1.0.6
+        with:
+          toolchain: nightly
+          target: x86_64-unknown-linux-gnu
+          profile: minimal
+          components: rustfmt
+          default: true
+
+      - name: Test fmt
+        run: cargo dev fmt --check
+
+      # Restore the rust-toolchain-file
+      - name: Restore rust-toolchain-file
+        run: git checkout rust-toolchain
```

Comment:
I would move this right before  the `lintcheck` test. That way clippy_dev has to be only rebuild once.

Currently it is built three times:

1. For rustfmt with +nightly
2. For the other tests with +<rust-toolchain-nightly>
3. For lintcheck --feature lintcheck